foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/OrphedomosToFabric8Migration.java

```diff
@@ -0,0 +1,283 @@
+    private static final String ARTIFACT_ID = "artifactId";
+    private static String imageName;
+    private static String imageVersion;
+
+    /**
+     * Function that returns instance of Plugin class with name equal to the pluginName provided
+     * @param model the model to check
+     * @param pluginName the name of the plugin
+     * @return Plugin with the plugin name specified
+     */
+    private Plugin getPlugin(Model model, String pluginName) {
+        Build build = model.getBuild();
+        if (build != null) {
+            List<Plugin> plugins = build.getPlugins();
+            for (Plugin plugin : plugins) {
+                if ((pluginName).equalsIgnoreCase(plugin.getArtifactId())) {
+                    return plugin;
+                }
+            }
+        }
+        return null;
+    }
+
+    private boolean isRootProjectPom(Model model) {
+        return ROOT_GROUP_ID.equals(model.getParent().getGroupId()) && ROOT_ARTIFACT_ID.equals(model.getParent().getArtifactId());
+    }
+    private boolean isTestDockerPom(Model model){
+        return model.getParent().getArtifactId().endsWith(TESTS_DOCKER_PARENT_ARTIFACT_ID) && model.getArtifactId().endsWith(TESTS_DOCKER_ARTIFACT_ID);
+    }
+
+    private Xpp3Dom getConfiguration(Plugin plugin) {
+        Object pluginConfiguration = plugin.getConfiguration();
+        return (Xpp3Dom) pluginConfiguration;
+    }
+
+    private static boolean hasConfigurationItem(Xpp3Dom configuration, String configurationItem){
+        return configuration.getChildren(configurationItem).length > 0;
+    }
+
+    /**
+     * Function to validate whether migration should be performed or not.
+     * @param file file to validate
+     * @return true if migration should be performed
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = false;
+        boolean migrateConfiguration = false;
+        boolean migratePackaging = false;
+
+        Model model = getLocationAnnotatedModel(file);
+        if (isRootProjectPom(model) || isTestDockerPom(model)){
+            shouldExecute = false;
+        } else {
+            Plugin orphedomosPlugin = getPlugin(model, ORPH_ARTIFACT_ID);
+            if (orphedomosPlugin != null){
+                migrateConfiguration = true;
+            }
+            if(containsOrphedomosPackaging(file)){
+                migratePackaging = true;
+            }
+            shouldExecute = migrateConfiguration || migratePackaging;
+        }
+
+        return shouldExecute;
+    }
+
+    /**
+     * Performs the migration if the shouldExecuteOnFile() returns true
+     * @param file - file to migrate
+     * @return isSuccessful - whether file was migrated successfully or not
+     */
+    @Override
+    protected boolean performMigration(File file) {
+        Model model = getLocationAnnotatedModel(file);
+        Plugin plugin  = getPlugin(model, ORPH_ARTIFACT_ID);
```

Comment:
If getPlugin() throws an exception instead of a null, then you can just wrap it all in a try catch and remove the null checking.